src/Orleans/Logging/TraceLogger.cs

```diff
@@ -287,6 +315,20 @@ public static void Initialize(ITraceConfiguration config, bool configChange = fa
                     // Plumb in log consumer to write to Trace listeners
                     var traceLogConsumer = new LogWriterToTrace();
                     LogConsumers.Add(traceLogConsumer);
+
+                    bool containsTraceTelemetryConsumer = false;
```

Comment:
Looks like we can avoid this three times code duplication with a helper function that would take the type we are looking for.